plugins/modules/idrac_license.py

```diff
@@ -0,0 +1,1090 @@
+
+        :return: The license URL as a string.
+        """
+        v1_resp = get_dynamic_uri(self.idrac, REDFISH)
+        license_service_url = v1_resp.get('LicenseService', {}).get(ODATA, {})
+        license_service_resp = get_dynamic_uri(self.idrac, license_service_url)
+        license_url = license_service_resp.get('Licenses', {}).get(ODATA, {})
+        return license_url
+
+    def get_job_status(self, module, license_job_response):
+        """
+        Get the status of a job.
+
+        Args:
+            module (object): The module object.
+            license_job_response (object): The response object for the license job.
+
+        Returns:
+            dict: The job details.
+        """
+        res_uri = validate_and_get_first_resource_id_uri(self.module, self.idrac, MANAGERS_URI)
+        job_tracking_uri = license_job_response.headers.get("Location")
+        job_id = job_tracking_uri.split("/")[-1]
+        job_uri = IDRAC_JOB_URI.format(job_id=job_id, res_uri=res_uri[0])
+        job_failed, msg, job_dict, wait_time = idrac_redfish_job_tracking(self.idrac, job_uri)
+        job_dict = remove_key(job_dict, regex_pattern='(.*?)@odata')
+        if job_failed:
+            module.exit_json(
+                msg=job_dict.get('Message'),
+                failed=True,
+                job_details=job_dict)
+        return job_dict
+
+    def get_share_details(self, module):
+        """
+        Retrieves the share details from the given module.
+
+        Args:
+            module (object): The module object containing the share parameters.
+
+        Returns:
+            dict: A dictionary containing the share details with the following keys:
+                - IPAddress (str): The IP address of the share.
+                - ShareName (str): The name of the share.
+                - UserName (str): The username for accessing the share.
+                - Password (str): The password for accessing the share.
+        """
+        share_details = {}
+        share_details["IPAddress"] = module.params.get('share_parameters').get('ip_address')
+        share_details["ShareName"] = module.params.get('share_parameters').get('share_name')
+        share_details["UserName"] = module.params.get('share_parameters').get('username')
+        share_details["Password"] = module.params.get('share_parameters').get('password')
+        return share_details
+
+    def get_proxy_details(self, module):
+        """
+        Retrieves the proxy details based on the provided module parameters.
+
+        Args:
+            self: The instance of the class.
+            module: The module object containing the parameters.
+
+        Returns:
+            dict: A dictionary containing the proxy details.
+        """
+        proxy_details = {}
+        proxy_details["ShareType"] = module.params.get('share_parameters').get('share_type').upper()
+        proxy_details["IPAddress"] = module.params.get('share_parameters').get('ip_address')
```

Comment:
line 541-544 is common with get_share_details, utilize that

Reply:
Done